server/datastore/mysql/software.go

```diff
@@ -3264,11 +3267,14 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 						// we want to treat the installed vpp app as a regular software title
 						delete(bySoftwareTitleID, s.ID)
 					}
+					byVPPAdamID[*s.VPPAppAdamID] = s
 				} else {
 					continue
 				}
+			} else if opts.OnlyAvailableForInstall || opts.IncludeAvailableForInstall {
+				byVPPAdamID[*s.VPPAppAdamID] = s
 			}
-			byVPPAdamID[*s.VPPAppAdamID] = s
+			vppQuarantineSet[*s.VPPAppAdamID] = s
```

Comment:
Why did we need to add this (and the behavior on L3289/etc.?

Reply:
Later on down we put the app back into `byVPPAdamID` if we detected it to be installed (Line 3300). I cannot remember why `hostInstalledVppsApps` doesn't add records into `byVPPAdamID`. I think it might have something to do with the records being returned by `hostInstalledSoftware`.